cli/templatepull_test.go

```diff
@@ -230,118 +230,111 @@ func TestTemplatePull_LatestStdout(t *testing.T) {
-	source2 := genTemplateVersionSource()
-
-	expected, err := echo.Tar(source2)
-	require.NoError(t, err)
-
-	version1 := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, source1)
-	_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, version1.ID)
-
-	template := coderdtest.CreateTemplate(t, client, owner.OrganizationID, version1.ID)
-
-	// Update the template version so that we can assert that templates
-	// are being sorted correctly.
-	updatedVersion := coderdtest.UpdateTemplateVersion(t, client, owner.OrganizationID, source2, template.ID)
-	_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, updatedVersion.ID)
-	coderdtest.UpdateActiveTemplateVersion(t, client, template.ID, updatedVersion.ID)
-
-	dir := t.TempDir()
-
-	expectedDest := filepath.Join(dir, "expected")
-	actualDest := filepath.Join(dir, "actual")
-	ctx := context.Background()
-
-	err = extract.Tar(ctx, bytes.NewReader(expected), expectedDest, nil)
-	require.NoError(t, err)
-
-	inv, root := clitest.New(t, "templates", "pull", template.Name, actualDest)
-	clitest.SetupConfig(t, templateAdmin, root)
+	// Prevents the tests from running in parallel.
+	tmp := t.TempDir()
+	expectedDest := filepath.Join(tmp, "expected")
+
+	tests := []struct {
+		name      string
+		givenPath string
+	}{
+		{
+			name:      "absolute path works",
+			givenPath: filepath.Join(tmp, "actual"),
+		},
+		{
+			name:      "relative path to specific dir is sanitized",
+			givenPath: "./pulltmp",
+		},
+		{
+			name:      "relative path to current dir is sanitized",
+			givenPath: ".",
+		},
+		{
+			name:      "directory traversal is acceptable",
+			givenPath: "../mytmpl",
+		},
+		{
+			name:      "empty path falls back to using template name",
+			givenPath: "",
+		},
+	}
 
-	ptytest.New(t).Attach(inv)
+	// nolint: paralleltest // These tests all share expectedDest
+	for _, tc := range tests {
+		tc := tc
 
-	require.NoError(t, inv.Run())
+		t.Run(tc.name, func(t *testing.T) {
+			// Use a different working directory to not interfere with actual directory when using relative paths.
+			newWD := t.TempDir()
+			cwd, err := os.Getwd()
+			require.NoError(t, err)
+			require.NoError(t, os.Chdir(newWD))
 
-	require.Equal(t,
-		dirSum(t, expectedDest),
-		dirSum(t, actualDest),
-	)
-}
+			t.Cleanup(func() {
+				require.NoError(t, os.Chdir(cwd))
+			})
 
-// ToDir tests that 'templates pull' pulls down the active template and writes
-// it to a directory with the name of the template if the path is not implicitly
-// supplied.
-// nolint: paralleltest
-func TestTemplatePull_ToImplicit(t *testing.T) {
-	client := coderdtest.New(t, &coderdtest.Options{
-		IncludeProvisionerDaemon: true,
-	})
-	owner := coderdtest.CreateFirstUser(t, client)
-	templateAdmin, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID, rbac.RoleTemplateAdmin())
+			t.Cleanup(func() {
+				_ = os.RemoveAll(tc.givenPath)
+				_ = os.RemoveAll(expectedDest)
```

Comment:
We could perhaps combine these and chdir into a single cleanup? Or perhaps even better would be to implement `testutil.Chdir` since it's something we do every now and a then.